executor/inspection_profile.go

```diff
@@ -0,0 +1,479 @@
+			}
+			labelValue := fmt.Sprintf(" %.2fs", v)
+			pb.addEdge(n.getName(""), n.getName(label), labelValue, "", v)
+			labelValue = fmt.Sprintf("%s\n %.2fs (%.2f%%)", n.getName(label), v, v*100/pb.totalValue)
+			pb.addNodeDef(n.getName(label), labelValue, v, v)
+		}
+		weight = selfCost / 2
+		// Since this node has labels, all cost was consume on the children, so the selfCost is 0.
+		selfCost = 0
+	}
+
+	label := fmt.Sprintf("%s\n %.2fs (%.2f%%)\nof %.2fs (%.2f%%)",
+		name, selfCost, selfCost*100/pb.totalValue, nodeTotal, nodeTotal*100/pb.totalValue)
+	pb.addNodeDef(n.getName(""), label, weight, selfCost)
+}
+
+func (pb *profileBuilder) addNodeDef(name, labelValue string, fontWeight, colorWeight float64) {
+	baseFontSize, maxFontGrowth := 5, 18.0
+	fontSize := baseFontSize
+	fontSize += int(math.Ceil(maxFontGrowth * math.Sqrt(math.Abs(fontWeight)/pb.totalValue)))
+
+	pb.buf.WriteString(fmt.Sprintf(`N%d [label="%s" fontsize=%d shape=box color="%s" fillcolor="%s"]`,
+		pb.getNameID(name), labelValue, fontSize,
+		pb.dotColor(colorWeight/pb.totalValue, false),
+		pb.dotColor(colorWeight/pb.totalValue, true)))
+	pb.buf.WriteByte('\n')
+}
+
+func (pb *profileBuilder) addEdge(from, to, label, style string, value float64) {
+	weight := 1 + int(math.Min(value*100/pb.totalValue, 100))
+	color := pb.dotColor(value/pb.totalValue, false)
+	pb.buf.WriteString(fmt.Sprintf(`N%d -> N%d [`, pb.getNameID(from), pb.getNameID(to)))
+	if label != "" {
+		pb.buf.WriteString(fmt.Sprintf(` label="%s" `, label))
+	}
+	if style != "" {
+		pb.buf.WriteString(fmt.Sprintf(` style="%s" `, style))
+	}
+	pb.buf.WriteString(fmt.Sprintf(` weight=%d color="%s"]`, weight, color))
+	pb.buf.WriteByte('\n')
+}
+
+func (pb *profileBuilder) ignoreFraction(value, total float64) bool {
+	return value*100/total < 0.01
+}
+
+// dotColor function is copy from https://github.com/google/pprof.
+func (pb *profileBuilder) dotColor(score float64, isBackground bool) string {
+	// A float between 0.0 and 1.0, indicating the extent to which
+	// colors should be shifted away from grey (to make positive and
+	// negative values easier to distinguish, and to make more use of
+	// the color range.)
+	const shift = 0.7
+	// Saturation and value (in hsv colorspace) for background colors.
+	const bgSaturation = 0.1
+	const bgValue = 0.93
+	// Saturation and value (in hsv colorspace) for foreground colors.
+	const fgSaturation = 1.0
+	const fgValue = 0.7
+	// Choose saturation and value based on isBackground.
+	var saturation float64
+	var value float64
+	if isBackground {
+		saturation = bgSaturation
+		value = bgValue
+	} else {
+		saturation = fgSaturation
+		value = fgValue
+	}
+
+	// Limit the score values to the range [-1.0, 1.0].
+	score = math.Max(-1.0, math.Min(1.0, score))
+
+	// Reduce saturation near score=0 (so it is colored grey, rather than yellow).
+	if math.Abs(score) < 0.2 {
+		saturation *= math.Abs(score) / 0.2
+	}
+
+	// Apply 'shift' to move scores away from 0.0 (grey).
```

Comment:
code here is quite complicated, better check it up yourself.

Reply:
This function returns the different color according to the score.